.github/workflows/main.yml

```diff
@@ -0,0 +1,342 @@
+        
+      - name: Commit Validation tests
+        run: ant -Dcluster.config=release commit-validation
+
+  linux-build-system-test:
+    needs: base-build
+    name: Build-System Test on Linux/JDK ${{ matrix.java }} 
+    runs-on: ubuntu-latest
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    strategy:
+      matrix:
+        java: [ '11' ]
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar -xzf build.tar.gz --exclude nbbuild/build/test
+
+      - name: Test Netbeans Build System
+        run: ant -Dcluster.config=release localtest
+
+  linux-javadoc:
+    needs: base-build
+    name: Build NBMs, Source zips and Javadoc on JDK ${{ matrix.java }} 
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '11' ]
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar -xzf build.tar.gz --exclude nbbuild/build/test
+
+      - name: Get maven coordinates
+        run: ant getallmavencoordinates
+
+      - name: Build nbms
+        run: ant build-nbms
+
+      - name: Build source zips
+        run: ant build-source-zips
+
+      - name: Build javadoc
+        run: ant build-javadoc
+
+  macos:
+    needs: base-build
+    name: Tests on MacOS/JDK ${{ matrix.java }} 
+    runs-on: macos-11
+    env:
+      ANT_OPTS: -Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json
+    strategy:
+      matrix:
+        java: [ '11' ]
+    steps:
+
+      - name: Set up JDK ${{ matrix.java }} 
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{ matrix.java }} 
+          distribution: 'zulu'
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/Library/Caches/Homebrew
+          key: ${{ runner.os }}-homebrew
+          restore-keys: ${{ runner.os }}-homebrew
+
+      - name: Download Build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+
+      - name: Extract
+        run: tar -xzf build.tar.gz --exclude nbbuild/build/test
+
+      - run: brew install ant
+
+      - name: Test platform/masterfs.macosx
+        run: ant -f platform/masterfs.macosx test
```

Comment:
Could we use _test distribution_ instead of the tar?

Reply:
potentially. What is the ant target for creating the test distribution? Some secondary jobs probably won't work with that though. E.g the nbm one.

Reply:
but let me ask first: is there a reason you prefer a test distribution over the full workspace approach? It is basically like testing a fresh clone, just from a snapshot which is already built.

My main argument against it is the artifact size. But I am not sure if that is a problem with 1 day retention size. Its likely that we will have to ask someone from apache infra anyway - even if we reduce the artifact size.